evalml/pipelines/component_graph.py

```diff
@@ -431,9 +441,10 @@ def _transform_features(
                 output_cache[f"{component_name}.x"] = output_x
                 output_cache[f"{component_name}.y"] = output_y
             else:
-                if fit:
+                if fit and (
+                    not component_instance._is_fitted or self.cached_data is None
```

Comment:
Same question here about why we need the `self.cached_data is None` check.

I just realized there was some previous discussion [here](https://github.com/alteryx/evalml/pull/3257#discussion_r817108930) so I commented there as well.

Reply:
Addressed in another comment!